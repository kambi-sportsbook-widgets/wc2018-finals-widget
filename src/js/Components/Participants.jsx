import React from "react";
import PropTypes from "prop-types";
import { translationModule } from "kambi-widget-core-library";
import DateComponent from "./Date";

const t = translationModule.getTranslation.bind(translationModule);
import styles from "./Participants.scss";

const WORLD_CUP_2018_ID = 2000075007;

const Participants = ({ event, flagBaseUrl, iconUrl, onClick }) => {
  return (
    <div className={styles.participants} onClick={onClick}>
      <div className={styles.team}>
        <h2 className={styles.label}>{event.homeName}</h2>
      </div>
      <div className={styles.logo}>
        <h3 className={styles.label}>{t("championsLeague").toUpperCase()}</h3>
        <p className={styles.sublabel}>{t("leagueYear").toUpperCase()}</p>

        <DateComponent date={event.start} short={true} />
      </div>
      <div className={styles.team}>
        <h2 className={styles.label}>{event.awayName}</h2>
      </div>
    </div>
  );
};

Participants.propTypes = {
  event: PropTypes.object.isRequired,
  flagBaseUrl: PropTypes.string,
  iconUrl: PropTypes.string,
  onClick: PropTypes.func.isRequired
};

export default Participants;
