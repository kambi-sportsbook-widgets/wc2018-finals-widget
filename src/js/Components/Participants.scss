.participants {
  justify-content: center;
  cursor: pointer;
  position: relative;
  display: flex;
  flex-direction: row;
  flex-basis: 30%;
}

.participants-font-size {
  font-size: large;
}

.team {
  display: flex;
  flex-direction: column;
  justify-content: center;
  flex-basis: 30%;

  :global(.kw-custom-logo-large-type){
    background-position: center;
    background-repeat: no-repeat;
    flex-basis: 30%;
  }

  .label {
    text-align: center;
    font-size: 130%;
    font-weight: 500;
  }
}

.logo {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 0 1em;
  text-align: center;

  :global(.kw-custom-logo-large-type){
    background-position: center;
    background-repeat: no-repeat;
    flex-basis: 50%;
    height: 70px;
  }

  .label, .sublabel {
    margin: 0;
  }

  .label {
    margin-top: .5em;
  }

  .sublabel {
    font-size: 90%;
  }
}

@media screen and (max-width: 768px) {
  .participants {
    order: -1;
    flex-basis: 100%;
  }
}
