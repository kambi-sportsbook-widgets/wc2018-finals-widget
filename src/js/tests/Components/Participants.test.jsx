import React from 'react'
import { translationModule } from 'kambi-widget-core-library'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import Participants from '../../Components/Participants'

let renderer

const exampleEvent = {
  countriesEnglishNames: ['Country1', 'Country2'],
  homeName: 'Country1',
  awayName: 'Country2',
  englishName: 'Country1 - Country2',
  id: 1,
  start: 1528988400000,
  state: 'NOT_STARTED',
  type: 'ET_MATCH',
}

jest.mock('kambi-widget-core-library', () => ({
  translationModule: {
    getTranslation: jest.fn(key => `Translated ${key}`),
  },
}))

describe('Participants DOM rendering', () => {
  beforeEach(() => {
    renderer = new ReactShallowRenderer()
    translationModule.getTranslation = jest.fn(key => `Translated ${key}`)
  })

  it('renders correctly with an example event', () => {
    expect(
      renderer.render(<Participants event={exampleEvent} onClick={() => {}} />)
    ).toMatchSnapshot()
  })
})
