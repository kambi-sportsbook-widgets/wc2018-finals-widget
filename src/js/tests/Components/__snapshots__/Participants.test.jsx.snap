// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participants DOM rendering renders correctly with an example event 1`] = `
<div
  className="participants "
  onClick={[Function]}
>
  <div
    className="team"
  >
    <i />
    <h2
      className="label"
    >
      Country1
    </h2>
  </div>
  <div
    className="logo"
  >
    <i />
    <h3
      className="label"
    >
      TRANSLATED WC2018
    </h3>
    <p
      className="sublabel"
    >
      TRANSLATED RUSSIA
    </p>
    <DateComponent
      date={1528988400000}
      short={true}
    />
  </div>
  <div
    className="team"
  >
    <i />
    <h2
      className="label"
    >
      Country2
    </h2>
  </div>
</div>
`;
